exchanges/poloniex/poloniex_websocket.go

```diff
@@ -28,1128 +30,509 @@ import (
+	return strings.Join(splits[:length-2], "_"), intervalValue, err
+}
 
-	t, ok := notification[8].(string)
-	if !ok {
-		return fmt.Errorf("%w time not string", errTypeAssertionFailure)
-	}
-	timeParse, err := time.Parse(time.DateTime, t)
+// Subscribe sends a websocket message to receive data from the channel
+func (e *Exchange) Subscribe(ctx context.Context, conn websocket.Connection, subs subscription.List) error {
+	subs, err := subs.ExpandTemplates(e)
 	if err != nil {
 		return err
 	}
+	return e.manageSubs(ctx, "subscribe", conn, subs, e.handleSubscription, &e.spotSubMtx)
+}
 
-	// null returned so ok check is not needed
-	clientOrderID, _ := notification[9].(string)
-
-	tt, ok := notification[10].(string)
-	if !ok {
-		return fmt.Errorf("%w time not string", errTypeAssertionFailure)
-	}
-	tradeTotal, err := strconv.ParseFloat(tt, 64)
+// Unsubscribe sends a websocket message to stop receiving data from the channel
+func (e *Exchange) Unsubscribe(ctx context.Context, conn websocket.Connection, subs subscription.List) error {
+	subs, err := subs.ExpandTemplates(e)
 	if err != nil {
 		return err
 	}
-
-	e.Websocket.DataHandler <- &order.Detail{
-		Exchange: e.Name,
-		OrderID:  strconv.FormatFloat(orderID, 'f', -1, 64),
-		Fee:      totalFee,
-		Trades: []order.TradeHistory{{
-			Price:     rate,
-			Amount:    amount,
-			Fee:       totalFee,
-			Exchange:  e.Name,
-			TID:       strconv.FormatFloat(tradeID, 'f', -1, 64),
-			Timestamp: timeParse,
-			Total:     tradeTotal,
-		}},
-		AssetType:     asset.Spot,
-		ClientOrderID: clientOrderID,
-	}
-	return nil
+	return e.manageSubs(ctx, "unsubscribe", conn, subs, e.handleSubscription, &e.spotSubMtx)
 }
 
-func (e *Exchange) processAccountKilledOrder(notification []any) error {
-	if len(notification) < 3 {
-		return errNotEnoughData
-	}
-
-	orderID, ok := notification[1].(float64)
-	if !ok {
-		return fmt.Errorf("%w order ID not float64", errTypeAssertionFailure)
-	}
-
-	// null returned so ok check is not needed
-	clientOrderID, _ := notification[2].(string)
-
-	e.Websocket.DataHandler <- &order.Detail{
-		Exchange:      e.Name,
-		OrderID:       strconv.FormatFloat(orderID, 'f', -1, 64),
-		Status:        order.Cancelled,
-		AssetType:     asset.Spot,
-		ClientOrderID: clientOrderID,
+func (e *Exchange) manageSubs(ctx context.Context, operation string, conn websocket.Connection, subscs subscription.List, payloadGen func(op string, s *subscription.Subscription) (*SubscriptionPayload, error), sendMsgLock *sync.Mutex) error {
+	var errs error
+	for _, s := range subscs {
+		payload, err := payloadGen(operation, s)
+		if err != nil {
+			errs = common.AppendError(errs, err)
+			continue
+		}
+		sendMsgLock.Lock()
+		result, err := conn.SendMessageReturnResponse(ctx, request.UnAuth, "subscription", &payload)
+		sendMsgLock.Unlock()
+		if err != nil {
+			errs = common.AppendError(errs, fmt.Errorf("%w %w subscribing to channel: %s", websocket.ErrSubscriptionFailure, err, payload.Channel[0]))
```

Comment:
🚧 Change request:
Again: You cannot say `subscribing` here.
I provided both a comment and also a provided a patch with the correct wording 😢 
```suggestion
			errs = common.AppendError(errs, fmt.Errorf("%w %s channel %q: %w", websocket.ErrSubscriptionFailure, operation, payload.Channel[0], err))
```